pkg/builder/merged_entity_meta_data_builder.go

```diff
@@ -267,45 +298,100 @@ func (builder *MergedEntityMetadataBuilder) InternalMatchingType(returnType Retu
 }
 
 func (builder *MergedEntityMetadataBuilder) InternalMatchingProperty(propertyName string) *MergedEntityMetadataBuilder {
-	internal := &matchingData{
-		propertyName: propertyName,
-	}
+	internal := createMatchingData(propertyName, "", []string{}, "")
```

Comment:
I feel, though not very strongly, that the old way is easier to read than passing in multiple empty-like arguments.